### PR TITLE
Improve resilience of autoscaler 12 nodes total test

### PR DESCRIPTION
**Pull request by JoelSpeed**
This is an attempt to improve the resilience of the following test. In this case, the node had gone away but the Machine had not. We can check those two things separately. The `GetNodesFromMachineSet` helper in the framework should be ignoring the not found error, it is up to the caller to check the number is correct.

```
Machine Suite: [Feature:Machines] Autoscaler should use a ClusterAutoscaler that has 12 maximum total nodes count and balance similar nodes enabled scales up and down while respecting MaxNodesTotal [Slow][Serial] expand_less | 23m20s
-- | --
/go/src/github.com/openshift/cluster-api-actuator-pkg/pkg/autoscaler/autoscaler.go:521 Timed out after 360.001s. Error: Unexpected non-nil/non-zero extra argument at index 1: 	<*fmt.wrapError>: &fmt.wrapError{msg:"error getting node from machine \"ci-op-cr8lz3b0-d5662-pr6wdmng5s-27ltb\": nodes \"ci-op-cr8lz3b0-d5662-pr6wdmng5s-27ltb\" not found", err:(*errors.StatusError)(0xc00065c320)} /go/src/github.com/openshift/cluster-api-actuator-pkg/pkg/autoscaler/autoscaler.go:589
```

### CODE REVIEW

**Review by elmiko — APPROVED** [review]
thanks Joel!
/approve

File: pkg/framework/nodes.go [hunk 1]
```diff
@@ -71,7 +72,11 @@ func GetNodesFromMachineSet(client runtimeclient.Client, machineSet *mapiv1beta1
 	var nodes []*corev1.Node
 	for key := range machines {
 		node, err := GetNodeForMachine(client, machines[key])
-		if err != nil {
+		if err != nil && apierrors.IsNotFound(err) {
```
Comment (Danil-Grigorev) [inline pkg/framework/nodes.go:75]: Just `apierrors.IsNotFound(err)` is enough.

File: pkg/framework/nodes.go [hunk 2]
```diff
@@ -71,7 +72,11 @@ func GetNodesFromMachineSet(client runtimeclient.Client, machineSet *mapiv1beta1
...
+			// We don't care about not found errors.
+			// Callers should account for the number of nodes being correct or not.
```
Comment (Danil-Grigorev) [inline pkg/framework/nodes.go:77]: It still will be better to put a log message in there. The method signature suggests it should be a matching list to the assigned Machines.
```golang
klog.Infof("No Node object found for machine %s", machine.GetName())
```

**Review by Danil-Grigorev — COMMENTED** [review]
Makes sense to me, just some comments.

**Comment by JoelSpeed** [comment]
@Danil-Grigorev Good feedback, I included your suggestions

**Pull request merged**